chain-abstraction/sol-implementation/edge_challenge_manager.go

```diff
@@ -328,62 +327,44 @@ func (e *specEdge) ConfirmByClaim(ctx context.Context, claimId protocol.ClaimId)
 // If two validators open a subchallenge S' at edge B in BigStepChallenge, the TopLevelClaimHeight
 // is the height of A.
 func (e *specEdge) TopLevelClaimHeight(ctx context.Context) (protocol.OriginHeights, error) {
```

Comment:
This function finds the claimed edge heights at each level above the current edge. For example:

say small step edge E has two challenge levels above it: big step and block challenge. This function would tell us that the claimed big step edge is at height 14 and the claimed block edge is at 3, for example